pkg/controller/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -386,6 +386,14 @@ func (r *ReconcileCredentialsRequest) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, nil
 	}
 
+	isStale := cr.Generation != cr.Status.LastSyncGeneration
+	hasRecentlySynced := cr.Status.LastSyncTimestamp != nil && cr.Status.LastSyncTimestamp.Add(time.Hour*1).After(time.Now())
```

Comment:
Not sure if you're just unit testing first but just in case, nothing is setting these yet in here that I can see, so you'll pass in tests but not in real world.